### PR TITLE
Improved compatibility with a spanner platform

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/NEX-543

### CODE REVIEW

File: common/persistence/class.SqlKvDriver.php [hunk 1]
```diff
@@ -74,54 +77,53 @@ public function set($id, $value, $ttl = null, $nx = false)
...
+
             if($platformName == 'mysql'){
                 //query found in Symfony PdoSessionHandler
                 $statement = "INSERT INTO kv_store (kv_id, kv_value, kv_time) VALUES (:id, :data, :time) 
                     ON DUPLICATE KEY UPDATE kv_value = VALUES(kv_value), kv_time = VALUES(kv_time)";
                 $returnValue = $this->sqlPeristence->exec($statement,$params);
-                
+
             } else if($platformName == 'oracle'){
                 $statement = "MERGE INTO kv_store USING DUAL ON(kv_id = :id)
```
Comment (julien-sebire) [inline common/persistence/class.SqlKvDriver.php]: Strange thing: for oracle, statement isn't executed. Is it ever used?

File: common/persistence/class.SqlKvDriver.php [hunk 2]
```diff
@@ -74,54 +77,53 @@ public function set($id, $value, $ttl = null, $nx = false)
...
-            }
-            
-            else {
+            } else {
                 $statement = 'UPDATE kv_store SET kv_value = :data , kv_time = :time WHERE kv_id = :id';
-                $returnValue = $this->sqlPeristence->exec($statement,$params);
-                if ($returnValue == 0){
-                    $returnValue = $this->sqlPeristence->insert('kv_store', array('kv_id' => $id, 'kv_time' => $expire, 'kv_value' => $encoded));
+                $returnValue = $this->sqlPeristence->exec($statement, $params, ['data' => ParameterType::STRING, 'time' => ParameterType::INTEGER, 'id' => ParameterType::STRING]);
+                if (0 == $returnValue) {
```
Comment (julien-sebire) [inline common/persistence/class.SqlKvDriver.php]: I'd rather see a strict comparison than weak with type juggling...

File: common/persistence/class.SqlKvDriver.php (hunk 1 above)
Comment (boajer) [reply common/persistence/class.SqlKvDriver.php]: I have doubts if oracle compatibility was tested recently, at all.

File: common/persistence/class.SqlKvDriver.php (hunk 2 above)
Comment (boajer) [reply common/persistence/class.SqlKvDriver.php]: Updated.

**Pull request merged**